platform/security/rest/security-rest-clientapi/src/main/java/org/codice/ddf/cxf/client/ClientBuilder.java

```diff
@@ -188,4 +188,12 @@
    * @return
    */
   ClientBuilder<T> useSamlEcp(boolean useSamlEcp);
+
+  /**
+   * True will enable the use of the SubjectRetrievalInterceptor for this client.
+   *
+   * @param useSubjectRetrievalInterceptor
+   * @return
+   */
+  ClientBuilder<T> useSubjectRetrievalInterceptor(boolean useSubjectRetrievalInterceptor);
```

Comment:
❓ In what case would you set this to false?

Reply:
Sorry, I'm still a little unclear here. What's the use case where the subject retrieval interceptor would _not_ be used?

Reply:
I don't have a very clear grasp on it either since I was focusing on preserving functionality. But I think I know its use.

What I know:
Lets say DDF1 subscribes to DDF2 through CSW by sending a subscription request. The subscription request contains a query and says "Hey, every time you (DDF2) update your catalog with an entry that matches the given query, I want you to send me (DDF1) those updates".

What I think I know:
Now when DDF2 updates is catalog appropriately, it needs to run the query and possible return results to DDF1. But that query might contain sensitive information that DDF1 isn't allowed to see, so before DDF2 runs the query it uses the SubjectRetreivalInterceptor to grab the certificates of DDF1, creates a subject, and runs the query as that subject in order to filter results DDF1 shouldn't see.

Reply:
The original javadoc says "stores the receivers subject in the header of the response with a key of ddf.security.Subject."

Not sure why we would care about the subject in a response.

Reply:
> Not sure why we would care about the subject in a response.

I don't get that either. Thanks for the explanation though 😄